### PR TITLE
feat(css): CSS is pre-processed by css-crush, all /cache output is filterable

### LINKED ISSUE #10625: adopt a css preprocessor
A pure php processor can be found here http://the-echoplex.net/csscrush

This would make css more readable and easier to maintain... also it would allow use to define some global vars to be used in css so they could be easier to replace in plugin (think of colors or font family).

### PR DESCRIPTION
**Pull request by mrclay**
All `/cache` output is now filtered by either the `simplecache:generate` or `cache:generate` hook, depending on whether Simplecache is enabled.

css-crush is used to pre-process all CSS `/cache` output.

Fixes #10625
Refs #10316

- [ ] test basic pre-processing

(replaces #10666)

### CODE REVIEW

**Comment by mrclay** [comment]
FYI with simplecache off, elgg.css takes ~200ms longer to load (or 304). This is still 140ms faster than the initial PR, and seems acceptable for development.

File: docs/guides/upgrading.rst [hunk 1]
```diff
@@ -401,6 +401,8 @@ Notable changes in plugins:
  * Support for ``.elgg-icon-hover`` has been dropped
  * Admin theme now reuses icon classes from ``elements/icons.css``
 
+Also note, CSS views served via ``/cache`` URLs are pre-processed using `CSS Crush <http://the-echoplex.net/csscrush/>`_. If you make references to CSS variables or other elements, the definition must be located within the same view output. E.g. A variable defined in ``elgg.css`` cannot be referenced in a separate CSS file like ``colorbox.css``.
+
```
Comment (jdalsem) [inline docs/guides/upgrading.rst]: in a future PR we will introduce a generic view with variables that gets injected into the csscrush api, this way you can use global (core) variables in all files..

**Review by jdalsem — APPROVED** [review]

**Comment by jdalsem** [comment]
@mrclay what kind of test would you like to add? As this PR is otherwise ready to be merged

**Comment by mrclay** [comment]
E.g. This removes vendor prefixes from our CSS so csscrush handles it. Would be nice to verify that happens.

**Comment by hypeJunction** [comment]
I am now leaning towards scraping the theme and building anew with Bootstrap 4. We keep reinventing the wheel. If that happens, I don't see a reason to use CSSCrush yet, as B4 uses Sass.

**Comment by jdalsem** [comment]
B4 'could' use SASS, but SASS and php do not work together so could. It would be cool to be able to preprocess css with a tool as a feature in Elgg. That way plugin devs can use css superglobals or mixins. The featurelist of CSS crush almost matches that of SASS, that is why i eventually suggested to use CSS Crush instead of SASS/LESS. There are some php-sass libraries, but those are not that complete. 

Theme builders can use sass for their own development of themes of course, but i would like to see preprocessing as part of Elgg core. The IMHO best solution for that is still CSS Crush

**Pull request merged**